tests/Application/Actions/Hooks/StripeUpdateTest.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MatchBot\Tests\Application\Actions\Hooks;
+
+use DI\Container;
+use Doctrine\ORM\EntityManagerInterface;
+use MatchBot\Application\Actions\ActionPayload;
+use MatchBot\Application\HttpModels\Donation as HttpDonation;
+use MatchBot\Domain\Donation;
+use MatchBot\Domain\DonationRepository;
+use MatchBot\Domain\StripeWebhook;
+use MatchBot\Tests\Application\Actions\DonationTestDataTrait;
+use MatchBot\Tests\TestCase;
+use Prophecy\Argument;
+use Slim\Exception\HttpNotFoundException;
+
+class StripeUpdateTest extends TestCase
+{
+    use DonationTestDataTrait;
+
+    public function testUnsupportedAction(): void
+    {
+        $app = $this->getAppInstance();
+        /** @var Container $container */
+        $container = $app->getContainer();
+
+        $body = file_get_contents(dirname(__DIR__, 3) . '/TestData/canceled.json');
+
+        $stripeRepoProphecy = $this->prophesize(StripeWebhook::class);
+        $entityManagerProphecy = $this->prophesize(EntityManagerInterface::class);
+        $donationRepoProphecy = $this->prophesize(DonationRepository::class);
+
+        $container->set(StripeWebhook::class, $stripeRepoProphecy->reveal());
+        $container->set(EntityManagerInterface::class, $entityManagerProphecy->reveal());
+        $container->set(DonationRepository::class, $donationRepoProphecy->reveal());
+
+        $stripeSignature = 't=' . time() . 'v1=' . $this->getValidAuth($body) . 'v0=' . $this->getValidAuth($body);
+
+        $request = $this->createRequest('POST', '/hooks/stripe', $body)
+            ->withHeader('stripe-signature', $stripeSignature);
+        
+        $response = $app->handle($request);
```

Comment:
@NoelLH any idea where I'm going wrong here, the test is throwing an error `Using $this when not in object context` - here's the stack trace:

> 1) MatchBot\Tests\Application\Actions\Hooks\StripeUpdateTest::testUnsupportedAction
Error: Using $this when not in object context

/var/www/html/src/Application/Actions/Hooks/StripeUpdate.php:53
/var/www/html/src/Application/Actions/Action.php:44
/var/www/html/vendor/slim/slim/Slim/Handlers/Strategies/RequestResponse.php:43
/var/www/html/vendor/slim/slim/Slim/Routing/Route.php:381
/var/www/html/vendor/slim/slim/Slim/MiddlewareDispatcher.php:81
/var/www/html/vendor/slim/slim/Slim/MiddlewareDispatcher.php:81
/var/www/html/vendor/slim/slim/Slim/Routing/Route.php:341
/var/www/html/vendor/slim/slim/Slim/Routing/RouteRunner.php:84
/var/www/html/app/routes.php:40
/var/www/html/vendor/slim/slim/Slim/MiddlewareDispatcher.php:283
/var/www/html/vendor/slim/slim/Slim/Middleware/RoutingMiddleware.php:60
/var/www/html/vendor/slim/slim/Slim/MiddlewareDispatcher.php:140
/var/www/html/vendor/slim/slim/Slim/MiddlewareDispatcher.php:81
/var/www/html/vendor/slim/slim/Slim/App.php:215
/var/www/html/tests/Application/Actions/Hooks/StripeUpdateTest.php:44

Reply:
Ignore this, got it working in the end!